module/zfs/arc.c

```diff
@@ -4735,7 +4735,7 @@ arc_evict_cb(void *arg, zthr_t *zthr)
 	 * broadcast will wake any remaining arc evict waiters.
 	 */
 	mutex_enter(&arc_evict_lock);
-	arc_evict_needed = !zthr_iscancelled(arc_evict_zthr) &&
+	arc_evict_needed = !zthr_iscancelled(zthr) &&
```

Comment:
This change come from here: https://github.com/openzfs/zfs/pull/12225/commits/c32cc85fd0c0bb6eb479086e9e747bcc3fbe5f8b
It is needed to cancel thread thru zthr pointer because arc_evict_zthr or spa_raidz_expand_zthr could be NULL.
It can happen, if current thread starts earlier than zthr_create() returned, like in spa_start_raidz_expansion_thread().

Reply:
But since this is a bit surprising, a comment describing the situation should be added to the code.

Reply:
I added a comment to explain why the passed in `zthr` is preferred.